amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -155,14 +210,84 @@ describe('SearchPage', () => {
     });
 
     afterAll(() => {
-      getSelectedTabByResourceTypeSpy.mockRestore();
       createSearchOptionsSpy.mockRestore();
     });
   });
 
+  describe('componentDidUpdate', () => {
+    let props;
+    let wrapper;
+    let searchAllSpy;
+    beforeAll(() => {
+      const setupResult = setup({
+        location: {
+          search: '/search?searchTerm=current&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      });
+      props = setupResult.props;
+      wrapper = setupResult.wrapper;
+      const mockPrevProps = {
+        location: {
+          search: '/search?searchTerm=previous2&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      };
+      searchAllSpy = jest.spyOn(props, 'searchAll');
```

Comment:
Because `setStateSpy` is reused between tests, we should call `setStateSpy.mockClear();` before these blocks of test run. 

There actually multiple options here:
1. Create a new `setStateSpy` for every test
2. Call `mockClear()` before every test block that uses `setStateSpy`
3. Call `mockClear()` after every test block that uses `setStateSpy`
4. Some people do both 2 & 3 to be safe.

At the moment I have personally opted for Option 2 to create some consistency, but we need to make sure we do at least one of those options to prevent bugs in our tests.